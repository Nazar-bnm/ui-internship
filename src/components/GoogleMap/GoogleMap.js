import React, { Component } from 'react';
import cx from 'classnames';
import PropTypes from 'prop-types';

import './GoogleMap.scss';

export const CN = 'google-map';

const locations = [
  ['Forum Lviv', 49.8499054, 24.0201103, 3],
  ['Viktoria Gardens', 49.807371, 23.975852, 2],
  ['King Cross', 49.7738766, 24.0086945, 1]
];

class GoogleMap extends Component {
  constructor(props) {
    super(props);

    this.onPageLoad = this.onPageLoad.bind(this);
  }

  componentDidMount() {
    if (!window.google) {
      const script = document.createElement('script');
      script.src = 'https://maps.google.com/maps/api/js?key=AIzaSyAyesbQMyKVVbBgKVi2g6VX7mop2z96jBo';
      script.id = 'googleMaps';
      document.body.appendChild(script);
      script.addEventListener('load', () => this.onPageLoad());
    } else {
      this.onPageLoad();
    }
  }

  onPageLoad() {
    const { id, options } = this.props;
    const map = new window.google.maps.Map(document.getElementById(id), options);
    const infowindow = new window.google.maps.InfoWindow();

    for (let i = 0; i < locations.length; i++) {
      const marker = new window.google.maps.Marker({
        position: new window.google.maps.LatLng(locations[i][1], locations[i][2]),
        map
      });
      // const event = new window.google.maps.event.addListener(marker, 'click', (function renderMap() {
        //   infowindow.setContent(locations[i][0]);
        //   infowindow.open(map, marker);
        // }(marker, i)));
        // // event.addListener(marker, 'click', (function renderMap() {
      // //     infowindow.setContent(locations[i][0]);
      // //     infowindow.open(map, marker);
      // // }(marker, i)));
      // eslint-disable-next-line
      const test = () => new window.google.maps.event.addListener(marker, 'click', (function renderMarker() {
        return function renderMap() {
          infowindow.setContent(locations[i][0]);
          infowindow.open(map, marker);
        };
      }(marker, i)));
      test();
    }
  }


  render() {
    const { id } = this.props;
    return <div id={id} className={cx(CN)} />;
  }
}

GoogleMap.propTypes = {
  id: PropTypes.string.isRequired,
  options: PropTypes.object.isRequired
};

export default GoogleMap;
