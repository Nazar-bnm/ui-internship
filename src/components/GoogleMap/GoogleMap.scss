.google-map {
  width: 100%;
  height: 100px;

  &__modal {
    height: 400px;
  }
}

