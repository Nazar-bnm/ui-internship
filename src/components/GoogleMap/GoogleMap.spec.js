import React from 'react';
import { shallow, mount } from 'enzyme';

import GoogleMap from './GoogleMap';

describe('<GoogleMap />', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      id: 'myMap',
      options: { center: { lat: 49.8397, lng: 24.0297 }, zoom: 11 }
    }

    wrapper = shallow(<GoogleMap {...props}  />)
  });

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should check onPageLoad()', () => {
    const inst = wrapper.instance();
    const spy = jest.spyOn(inst, 'onPageLoad');

    inst.onPageLoad();
    expect(spy).toHaveBeenCalled();
  });

  xtest('should check renderMap()', () => {
    const inst = wrapper.instance();
    const spy = jest.spyOn(inst, 'renderMap');

    inst.renderMap();
    expect(spy).toHaveBeenCalled();
  });

});
