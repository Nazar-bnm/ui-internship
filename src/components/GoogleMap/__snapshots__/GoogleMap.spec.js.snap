// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GoogleMap /> should match snapshot 1`] = `
<div>
  <div
    className="google-map"
    id="myMap"
    onClick={[Function]}
  />
</div>
`;
